pkg/api/store/cluster/cluster_store.go

```diff
@@ -405,3 +406,30 @@ func setBackupConfigSecretKeyIfNotExists(oldData, newData map[string]interface{}
 		values.PutValue(newData, oldSecretKey, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig", "s3BackupConfig", "secretKey")
 	}
 }
+
+func setPrivateRegistryPasswordIfNotExists(oldData, newData map[string]interface{}) {
+	newSlice, ok := values.GetSlice(newData, "rancherKubernetesEngineConfig", "privateRegistries")
+	if !ok || newSlice == nil {
+		return
+	}
+	oldSlice, ok := values.GetSlice(oldData, "rancherKubernetesEngineConfig", "privateRegistries")
+	if !ok || oldSlice == nil {
+		return
+	}
+
+	var updatedConfig []map[string]interface{}
+	for _, newConfig := range newSlice {
+		if newConfig["password"] != nil {
+			updatedConfig = append(updatedConfig, newConfig)
+			continue
+		}
+		for _, oldConfig := range oldSlice {
+			if newConfig["url"] == oldConfig["url"] && oldConfig["password"] != nil {
```

Comment:
I think we need to compare `user` as well.

Reply:
Updated！